apps/angular/src/core/interceptors/auth-token.interceptor.ts

```diff
@@ -0,0 +1,117 @@
+import { HttpErrorResponse, HttpEvent, HttpHandlerFn, HttpRequest } from '@angular/common/http';
+import { DestroyRef, inject } from '@angular/core';
+import { BehaviorSubject, Observable, catchError, switchMap, throwError } from 'rxjs';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { AuthService } from '../services/auth.service';
+
+import { HttpErrors } from '../models/http-errors';
+import { LocalStorageService } from '../services/local-storage.service';
+
+const IS_REFRESHING$ = new BehaviorSubject(false);
+
+/**
+ * Add authorization header.
+ * @param req - Request.
+ * @param next - Next interceptor.
+ */
+export function authTokenInterceptor(req: HttpRequest<unknown>, next: HttpHandlerFn): Observable<HttpEvent<unknown>> {
+	const localStorageService = inject(LocalStorageService);
+	const authService = inject(AuthService);
+	const destroyRef = inject(DestroyRef);
+
+	const accessToken = localStorageService.getItem('accessToken');
+	const refreshToken = localStorageService.getItem('refreshToken');
+
+	let isRefreshing = false;
+
+	if (accessToken === null || accessToken.length === 0) {
+		return next(req);
+	}
+
+	IS_REFRESHING$
+		.pipe(takeUntilDestroyed(destroyRef))
+		.subscribe(value => {
+			isRefreshing = value;
+		});
+
+	if (isRefreshing) {
+		return handleRefresh({
+			authService,
+			localStorageService,
+			request: req,
+			next,
+			refreshToken: refreshToken ?? '',
+			isRefreshing,
+		});
+	}
+
+	const authTokenRequest = addAuthorizationHeader(req, accessToken ?? '');
+	return next(authTokenRequest).pipe(
+		catchError((error: unknown) => {
+			const httpError = error as HttpErrorResponse;
+			if (httpError.status === HttpErrors.Unauthorized) {
+				return handleRefresh({
+					authService,
+					localStorageService,
+					request: req,
+					next,
+					refreshToken: refreshToken ?? '',
+					isRefreshing,
+				});
+			}
+			return throwError(() => new Error('Something bad happened; please try again later.'));
```

Comment:
```suggestion
			return throwError(error);
```

You haven't actually handled other errors, you shouldn't suppress them this way. Let someone else handle this error.